src/handlers/web3.ts

```diff
@@ -698,53 +703,51 @@ export const getDataForNftTransfer = async (
   from: string,
   to: string,
   asset: ParsedAddressAsset
-): Promise<string> => {
-  const nftVersion = asset.asset_contract?.nft_version;
-  const schema_name = asset.asset_contract?.schema_name;
-  if (nftVersion === '3.0') {
-    const transferMethodHash = smartContractMethods.nft_transfer_from.hash;
-    const data = ethereumUtils.getDataString(transferMethodHash, [
-      ethereumUtils.removeHexPrefix(from),
+): Promise<string | undefined> => {
+  if (!asset.id || !asset.asset_contract?.address) return;
+  const lowercasedContractAddress = asset.asset_contract.address.toLowerCase();
+  const standard = asset.asset_contract?.schema_name;
+  let data;
+  if (
+    lowercasedContractAddress === CRYPTO_KITTIES_NFT_ADDRESS &&
+    asset.network === Network.mainnet
+  ) {
+    const transferMethod = smartContractMethods.token_transfer;
+    data = ethereumUtils.getDataString(transferMethod.hash, [
       ethereumUtils.removeHexPrefix(to),
       convertStringToHex(asset.id),
     ]);
-    return data;
-  } else if (schema_name === 'ERC1155') {
-    const transferMethodHash =
-      smartContractMethods.erc1155_safe_transfer_from.hash;
-    const data = ethereumUtils.getDataString(transferMethodHash, [
+  } else if (
+    lowercasedContractAddress === CRYPTO_PUNKS_NFT_ADDRESS &&
+    asset.network === Network.mainnet
+  ) {
+    const transferMethod = smartContractMethods.punk_transfer;
+    data = ethereumUtils.getDataString(transferMethod.hash, [
+      ethereumUtils.removeHexPrefix(to),
+      convertStringToHex(asset.id),
+    ]);
+  } else if (standard === ERC1155) {
+    const transferMethodHash = smartContractMethods.erc1155_transfer.hash;
+    data = ethereumUtils.getDataString(transferMethodHash, [
       ethereumUtils.removeHexPrefix(from),
       ethereumUtils.removeHexPrefix(to),
       convertStringToHex(asset.id),
       convertStringToHex('1'),
       convertStringToHex('160'),
       convertStringToHex('0'),
     ]);
-    return data;
-  } else if (IS_TESTING === 'true') {
-    const transferMethodHash = smartContractMethods.nft_transfer.hash;
-    const data = ethereumUtils.getDataString(transferMethodHash, [
+  } else {
+    // ERC721s + unsupported NFTs (will most likely fail)
```

Comment:
no harm in trying to send an unsupported nft here. very high chance of failing (will not charge gas, it actually fails on the gas estimation step). will only succeed if the nft just so happens to support the correct transfer method